src/FSharpVSPowerTools.Logic/DepthTagger.fs

```diff
@@ -30,10 +39,15 @@ type DepthTagger(buffer: ITextBuffer, filename: string, languageService: VSLangu
     
     let refreshTags() = 
         let uiContext = SynchronizationContext.Current
-        async { 
+        asyncMaybe { 
             let snapshot = buffer.CurrentSnapshot // this is the possibly-out-of-date snapshot everyone here works with
-            let source = snapshot.GetText() 
-            let! ranges = languageService.CheckerAsync (DepthParser.getNonoverlappingDepthRanges source filename)
+            let source = snapshot.GetText()
+            let dte = serviceProvider.GetService<EnvDTE.DTE, SDTE>()
+            let! doc = dte.GetCurrentDocument doc.FilePath
+            let! project = projectFactory.CreateForDocument buffer doc
+            let! opts = languageService.GetProjectCheckerOptions project |> liftAsync
```

Comment:
Are the results of this saved?  Or are you recomputing the options on each change of the document? Just checking.